### ISSUE TITLE
readthedocs api autodoc builds is broken

### ISSUE THREAD

**Issue by wbolster** [wbolster/happybase#121]
snippet from readthedocs build log:

```
python /home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/bin/sphinx-build -t -e -b readthedocs -d _build/doctrees-readthedocs -d language=en . _build/html
running sphinx v1.3.5
making output directory...
loading translations [en]... done
building [mo]: targets for 0 po files that are out of date
building [readthedocs]: targets for 9 source files that are out of date
updating environment: 9 added, 0 changed, 0 removed
reading sources... [ 11%] api
reading sources... [ 22%] development
reading sources... [ 33%] faq
reading sources... [ 44%] index
reading sources... [ 55%] installation
reading sources... [ 66%] license
reading sources... [ 77%] news
reading sources... [ 88%] todo
reading sources... [100%] user

/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/api.rst:36: warning: autodoc: failed to import class u'connection' from module u'happybase'; the following exception was raised:
traceback (most recent call last):
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  file "build/bdist.linux-x86_64/egg/happybase/__init__.py", line 11, in <module>
    from .connection import default_host, default_port, connection
  file "build/bdist.linux-x86_64/egg/happybase/connection.py", line 13, in <module>
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/hook.py", line 24, in load_module
    return load_module(fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 75, in load_module
    module = load(thrift_file, module_name=fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 30, in load
    include_dir=include_dir)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/parser.py", line 487, in parse
    with open(path) as fh:
ioerror: [errno 20] not a directory: '/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/happybase-0.9-py2.7.egg/happybase/hbase.thrift'
/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/api.rst:42: warning: autodoc: failed to import class u'table' from module u'happybase'; the following exception was raised:
traceback (most recent call last):
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  file "build/bdist.linux-x86_64/egg/happybase/__init__.py", line 11, in <module>
    from .connection import default_host, default_port, connection
  file "build/bdist.linux-x86_64/egg/happybase/connection.py", line 13, in <module>
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/hook.py", line 24, in load_module
    return load_module(fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 75, in load_module
    module = load(thrift_file, module_name=fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 30, in load
    include_dir=include_dir)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/parser.py", line 487, in parse
    with open(path) as fh:
ioerror: [errno 20] not a directory: '/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/happybase-0.9-py2.7.egg/happybase/hbase.thrift'
/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/api.rst:48: warning: autodoc: failed to import class u'batch' from module u'happybase'; the following exception was raised:
traceback (most recent call last):
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  file "build/bdist.linux-x86_64/egg/happybase/__init__.py", line 11, in <module>
    from .connection import default_host, default_port, connection
  file "build/bdist.linux-x86_64/egg/happybase/connection.py", line 13, in <module>
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/hook.py", line 24, in load_module
    return load_module(fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 75, in load_module
    module = load(thrift_file, module_name=fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 30, in load
    include_dir=include_dir)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/parser.py", line 487, in parse
    with open(path) as fh:
ioerror: [errno 20] not a directory: '/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/happybase-0.9-py2.7.egg/happybase/hbase.thrift'
/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/api.rst:54: warning: autodoc: failed to import class u'connectionpool' from module u'happybase'; the following exception was raised:
traceback (most recent call last):
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  file "build/bdist.linux-x86_64/egg/happybase/__init__.py", line 11, in <module>
    from .connection import default_host, default_port, connection
  file "build/bdist.linux-x86_64/egg/happybase/connection.py", line 13, in <module>
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/hook.py", line 24, in load_module
    return load_module(fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 75, in load_module
    module = load(thrift_file, module_name=fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 30, in load
    include_dir=include_dir)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/parser.py", line 487, in parse
    with open(path) as fh:
ioerror: [errno 20] not a directory: '/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/happybase-0.9-py2.7.egg/happybase/hbase.thrift'
/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/api.rst:56: warning: autodoc: failed to import class u'noconnectionsavailable' from module u'happybase'; the following exception was raised:
traceback (most recent call last):
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  file "build/bdist.linux-x86_64/egg/happybase/__init__.py", line 11, in <module>
    from .connection import default_host, default_port, connection
  file "build/bdist.linux-x86_64/egg/happybase/connection.py", line 13, in <module>
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/hook.py", line 24, in load_module
    return load_module(fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 75, in load_module
    module = load(thrift_file, module_name=fullname)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/__init__.py", line 30, in load
    include_dir=include_dir)
  file "/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/thriftpy/parser/parser.py", line 487, in parse
    with open(path) as fh:
ioerror: [errno 20] not a directory: '/home/docs/checkouts/readthedocs.org/user_builds/happybase/envs/latest/local/lib/python2.7/site-packages/happybase-0.9-py2.7.egg/happybase/hbase.thrift'
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 11%] api
writing output... [ 22%] development
writing output... [ 33%] faq
writing output... [ 44%] index
writing output... [ 55%] installation
writing output... [ 66%] license
writing output... [ 77%] news
writing output... [ 88%] todo
writing output... [100%] user

generating indices... genindex
writing additional pages... search
copying static files... warning: html_static_path entry u'/home/docs/checkouts/readthedocs.org/user_builds/happybase/checkouts/latest/doc/_static' does not exist
warning: html_static_path entry '/home/docs/checkouts/readthedocs.org/readthedocs/templates/sphinx/_static' does not exist
done
copying extra files... done
dumping search index in english (code: en) ... done
dumping object inventory... done
build succeeded, 7 warnings.
copying readthedocs-dynamic-include.js_t... done
copying readthedocs-data.js_t... done
copying searchtools.js_t... done
```

**Comment by wbolster** [comment]
this may have to do with the `thriftpy` import hook not working properly.

@ericholscher any ideas here? `thriftpy` adds an import hook so that `.thrift` files can be imported directly, but apparently that does not work during the build by rtd?

**Comment by wbolster** [comment]
fixed in 06108ce838727d415ec2cd7d01c8e93ada587ce0
